### PR TITLE
Hex Input Options

### PR DESCRIPTION
**Pull request by amm3**
For consideration, this commit does two things:
  * Automatically/always remove spaces in hex input
  * Add option (`-r, --revhex`) to reverse (by byte) hex input.  Useful when copy/pasting little-endian input.

### CODE REVIEW

File: epochalypse.py [hunk 1]
```diff
@@ -103,6 +103,9 @@ def main():
       help='Hexadecimal timemstamp value to be converted',
       metavar='')
```
Comment (pstirparo) [inline epochalypse.py]: Remove newline

File: epochalypse.py [hunk 2]
```diff
@@ -123,7 +126,10 @@ def main():
     fromEpoch(float(args.epoch_input))
     print('')
   elif args.hexadecimal_input:
-    epoch = fromHex(args.hexadecimal_input)
+    hex_text = args.hexadecimal_input.replace(' ', '')
+    if args.revhex:
+      hex_text = hex_text.decode('hex')[::-1].encode('hex')
```
Comment (pstirparo) [inline epochalypse.py]: That's python2, this new version of the script is python3 and decode method on string does not work.

**Review by pstirparo — CHANGES_REQUESTED** [review]
Thanks for the contribution, please find below some comments.

**Comment by amm3** [comment]
Good point, @pstirparo.  The only method I'm aware of that's compatible with both is to use binascii.  Py3 doesn't support `decode('hex')` and Py2 doesn't support `bytes.fromhex()`.  Does this work for you?

**Comment by amm3** [comment]
Have you had a chance to review my changes, @pstirparo. Github still shows "changes requested", but I believe I've addressed all your concerns.

File: epochalypse.py [hunk 3]
```diff
@@ -100,8 +101,10 @@ def main():
       help='Epoch time to be converted',
       metavar='')
   parser.add_argument('-x', '--hex', dest="hexadecimal_input", default=False,
-      help='Hexadecimal timemstamp value to be converted',
-      metavar='')
+      help='Hexadecimal timemstamp value to be converted', metavar='')
+
```
Comment (pstirparo) [inline epochalypse.py]: Delete new line

File: epochalypse.py [hunk 4]
```diff
@@ -100,8 +101,10 @@ def main():
...
+  parser.add_argument("-r", "--revhex", action="store_true", default=False,
```
Comment (pstirparo) [inline epochalypse.py]: Change double quote to single quote for consistency. Also in the line below.

File: epochalypse.py [hunk 5]
```diff
@@ -123,7 +126,10 @@ def main():
     fromEpoch(float(args.epoch_input))
     print('')
   elif args.hexadecimal_input:
-    epoch = fromHex(args.hexadecimal_input)
+    hex_text = args.hexadecimal_input.replace(' ', '')
+    if args.revhex:
+      hex_text = binascii.hexlify( binascii.unhexlify( hex_text )[::-1] ).decode()
```
Comment (pstirparo) [inline epochalypse.py]: Remove space within hexlify(), after the open brackets and before the closing ones

**Review by pstirparo — CHANGES_REQUESTED** [review]
This was long due, my apologies.
The code is working, just few "cosmetics" comments to be consistent with the code and then I will merge it.
Thanks again.

**Comment by pstirparo** [comment]
Thanks a lot for the contribution Adam :)

**Pull request merged**